File: IOS_swift/WitSDK/Sensor/Modular/Resolver/Protocol/IProtocolResolver.swift
//
//  协议解析器借口
//
//  Created by huangyajun on 2022/8/31.
//

import Foundation

protocol IProtocolResolver{
    
    // 发送数据
    func sendData(sendData:[UInt8], deviceModel:DeviceModel, waitTime:Int64) throws
    
    // 解算传感器主动会传的数据
    func passiveReceiveData(data:[UInt8], deviceModel:DeviceModel)
    
}

File: Android_Java/wit-example-ble5/app/src/main/java/com/wit/witsdk/sensor/modular/resolver/interfaces/IProtocolResolver.java
package com.wit.witsdk.sensor.modular.resolver.interfaces;

import com.wit.witsdk.sensor.modular.device.DeviceModel;

/**
 * 协议解析器接口
 *
 * @author huangyajun
 * @date 2022/4/21 11:25
 */
public interface IProtocolResolver {

    /**
     * 发送数据
     *
     * @author huangyajun
     * @date 2022/4/21 11:26
     */
    void sendData(byte[] sendData, DeviceModel deviceModel, int waitTime, ISendDataCallback callback);

    /**
     * 发送数据
     *
     * @author huangyajun
     * @date 2022/4/21 11:26
     */
    void sendData(byte[] sendData, DeviceModel deviceModel);

    /**
     * 解算被动接收的数据
     *
     * @author huangyajun
     * @date 2022/4/21 11:26
     */
    void passiveReceiveData(byte[] data, DeviceModel deviceModel);

}
